rtshell/rtwatch.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python2
+# -*- Python -*-
+# -*- coding: utf-8 -*-
+
+'''rtshell
+
+Copyright (C) 2009-2015
+    Geoffrey Biggs
+    RT-Synthesis Research Group
+    Intelligent Systems Research Institute,
+    National Institute of Advanced Industrial Science and Technology (AIST),
+    Japan
+    All rights reserved.
+Licensed under the GNU Lesser General Public License version 3.
+http://www.gnu.org/licenses/lgpl-3.0.en.html
+
+Implementation of the command to watch component events.
+
+'''
+
+from __future__ import print_function
+
+import argparse
+import os
+import os.path
+import rtctree.exceptions
+import rtctree.tree
+import rtctree.path
+import RTC
+import sys
+import traceback
+import time
+import atexit
+from threading import Event
+
+from rtshell import path
+from rtshell import rts_exceptions
+from rtctree.component import Component
+import rtshell
+
+STATUS_CODE_MAP = {
+    Component.INACTIVE: 'INACTIVE',
+    Component.ACTIVE: 'ACTIVE',
+    Component.ERROR: 'ERROR',
+    Component.UNKNOWN: 'UNKNOWN',
+    Component.CREATED: 'CREATED'
+}
+
+EC_EVENT_CODE_MAP = {
+    Component.EC_ATTACHED: 'EC_ATTACHED',
+    Component.EC_DETACHED: 'EC_DETACHED',
+    Component.EC_RATE_CHANGED: 'EC_RATE_CHANGED',
+    Component.EC_STARTUP: 'EC_STARTUP',
+    Component.EC_SHUTDOWN: 'EC_SHUTDOWN'
+}
+
+PORT_EVENT_CODE_MAP = {
+    Component.PORT_ADD: 'PORT_ADD',
+    Component.PORT_REMOVE: 'PORT_REMOVE',
+    Component.PORT_CONNECT: 'PORT_CONNECT',
+    Component.PORT_DISCONNECT: 'PORT_DISCONNECT'
+}
+
+CONFIG_EVENT_CODE_MAP = {
+    Component.CFG_UPDATE_SET: 'CFG_UPDATE_SET',
+    Component.CFG_UPDATE_PARAM: 'CFG_UPDATE_PARAM',
+    Component.CFG_SET_SET: 'CFG_SET_SET',
+    Component.CFG_ADD_SET: 'CFG_ADD_SET',
+    Component.CFG_REMOVE_SET: 'CFG_REMOVE_SET',
+    Component.CFG_ACTIVATE_SET: 'CFG_ACTIVATE_SET'
+}
+
+counter = 0
+event = Event()
```

Comment:
Rather than making `event` a global variable, it should be created in `print_logs()` and passed to each callback as an argument when the callback is registered using `rtc.add_callback()` (see the `add_callback()` documentation).